browser_patches/firefox/juggler/NetworkObserver.js

```diff
@@ -213,10 +284,13 @@ class NetworkRequest {
     // will always be 'PUT'. (from another place in the source base)
     newChannel.requestMethod = method || this.httpChannel.requestMethod;
 
+    this._responseInterceptor = new ResponseInterceptor(this, newChannel);
     this._networkObserver._responseInterceptionChannels.add(newChannel);
+    this._networkObserver._channelToRequest.set(newChannel, this);
```

Comment:
Let's comment why do we put something into _channelToRequest map. Is it to pick up future redirects?

Reply:
Done.